### ISSUE TITLE
[[help / maybe idea]] Is it possible to commit the annotations and highlights to the PDF?

### ISSUE THREAD

**Issue by korm** [elias-sundqvist/obsidian-annotator#25]
Would be helpful for portability if the highlights and annotations could be committed to the PDF.  Maybe that's not how ```hypothesis``` works, but would be excellent if feasible, or maybe I'm missing something that's already possible.

**Comment by elias-sundqvist** [comment]
I don't think that's possible with any existing solution. I'm not even sure if it's on the hypothes.is roadmap.  Maybe [this card](https://github.com/hypothesis/product-backlog/projects/6#card-12350986) is relevant, but it's marked as long-term.

Hypothes.is is designed to be able to annotate generic websites, so the annotations are stored in terms of things like number of characters and text content. Pdf annotations, on the other hand, are stored in terms of things like coordinates. So I don't think there is an easy conversion that can be done between the formats.

I could maybe try analyzing the rendered highlights and implementing something with [annotpdf](https://www.npmjs.com/package/annotpdf), but that would probably not be a small effort. 

I can add this to the todo list, but it won't be a near-term thing.

**Comment by korm** [comment]
No, don't worry about it.  I was a nice-to-have but not a must-have idea.

**Comment by wenlzhang** [comment]
The following plugin supports committing highlights to the original PDF file. Not sure if notes are supported though:
https://github.com/MarkMindCkm/obsidian-markmind/issues/15
